skyportal/tests/api/test_photometry.py

```diff
@@ -53,6 +55,404 @@ def test_token_user_post_get_photometry_data(
+    status, data = api(
+        'GET', f'photometry/{photometry_id}?format=both', token=upload_data_token
+    )
+    assert_api(status, data)
+
+    corrected_ref_flux = 8.01 / 10 ** (0.4 * (26.0 - 23.9))
+    corrected_ref_fluxerr = 0.01 / 10 ** (0.4 * (26.0 - 23.9))
+
+    assert np.isclose(data['data']['ref_flux'], corrected_ref_flux)
+    assert np.isclose(data['data']['ref_fluxerr'], corrected_ref_fluxerr)
+    assert np.isclose(data['data']['tot_flux'], corrected_ref_flux + corrected_flux)
+    assert np.isclose(
+        data['data']['tot_fluxerr'],
+        np.sqrt(corrected_fluxerr**2 + corrected_ref_fluxerr**2),
+    )
+
+    # patch the reference flux
+    status, data = api(
+        'PATCH',
+        f'photometry/{photometry_id}',
+        data={
+            'obj_id': str(public_source.id),
+            'mjd': 58003.0,
+            'instrument_id': ztf_camera.id,
+            'flux': 12.24,
+            'fluxerr': 0.031,
+            'zp': 25.0,
+            'magsys': 'ab',
+            'filter': 'ztfg',
+            'group_ids': [public_group.id],
+            'ref_flux': 9.02,
+            'ref_fluxerr': 0.03,
+            'ref_zp': 27.0,  # same zeropoint
+        },
+        token=upload_data_token,
+    )
+    assert_api(status, data)
+    status, data = api(
+        'GET', f'photometry/{photometry_id}?format=both', token=upload_data_token
+    )
+    assert_api(status, data)
+    corrected_ref_flux = 9.02 / 10 ** (0.4 * (27.0 - 23.9))
+    corrected_ref_fluxerr = 0.03 / 10 ** (0.4 * (27.0 - 23.9))
+
+    assert np.isclose(data['data']['ref_flux'], corrected_ref_flux)
+    assert np.isclose(data['data']['ref_fluxerr'], corrected_ref_fluxerr)
+    assert np.isclose(data['data']['tot_flux'], corrected_ref_flux + corrected_flux)
+    assert np.isclose(
+        data['data']['tot_fluxerr'],
+        np.sqrt(corrected_fluxerr**2 + corrected_ref_fluxerr**2),
+    )
+
+
+def test_ref_mag(upload_data_token, public_source, public_group, ztf_camera):
+    status, data = api(
+        'POST',
+        'photometry',
+        data={
+            'obj_id': str(public_source.id),
+            'mjd': 58003.0,
+            'instrument_id': ztf_camera.id,
+            'mag': 19.24,
+            'limiting_mag': 20.5,
+            'magerr': 0.123,
+            'magref': 17.01,
+            'e_magref': 0.01,
+            'magsys': 'ab',
+            'filter': 'ztfg',
+            'group_ids': [public_group.id],
+            'altdata': {'some_key': 'some_value'},
+        },
+        token=upload_data_token,
+    )
+    assert_api(status, data)
+
+    photometry_id = data['data']['ids'][0]
+    status, data = api(
+        'GET', f'photometry/{photometry_id}?format=both', token=upload_data_token
+    )
+    assert_api(status, data)
+
+    assert data['data']['altdata'] == {'some_key': 'some_value'}
+
+    expected_flux = 10 ** (-0.4 * (19.24 - 23.9))
+    expected_fluxerr = 0.123 / 2.5 * np.log(10) * expected_flux
```

Comment:
This happens to work but I think parentheses would make for a better read:
`0.123*(np.log(10)/2.5)*expected_flux`